incompatible with svelte 4?

Hi there. Seems like svelte-plausible-analytics has some problems when svelte runs in version 4.0.0 (as released today) when deploying to vercel:

[18:18:47.171] Running build in Washington, D.C., USA (East) – iad1
[18:18:47.240] Cloning github.com/ /sv-sb-patscen-auth-2023-03 (Branch: Features/aiReport, Commit: 1f409af)
[18:18:47.763] Cloning completed: 522.67ms
[18:18:51.293] Restored build cache
[18:18:51.340] Running "vercel build"
[18:18:51.864] Vercel CLI 30.2.3
[18:18:52.523] Installing dependencies...
[18:18:54.586] npm ERR! code ERESOLVE
[18:18:54.588] npm ERR! ERESOLVE unable to resolve dependency tree
[18:18:54.589] npm ERR! 
[18:18:54.589] npm ERR! While resolving: sk-supabase-auth@0.0.1
[18:18:54.589] npm ERR! Found: svelte@4.0.0
[18:18:54.589] npm ERR! node_modules/svelte
[18:18:54.589] npm ERR!   dev svelte@"^4.0.0" from the root project
[18:18:54.589] npm ERR!   peer svelte@"^3.54.0 || ^4.0.0-next.0" from @sveltejs/kit@1.20.4
[18:18:54.589] npm ERR!   node_modules/@sveltejs/kit
[18:18:54.590] npm ERR!     dev @sveltejs/kit@"^1.20.4" from the root project
[18:18:54.590] npm ERR!     peer @sveltejs/kit@"^1.20.2" from @accuser/svelte-plausible-analytics@0.4.0
[18:18:54.590] npm ERR!     node_modules/@accuser/svelte-plausible-analytics
[18:18:54.590] npm ERR!       @accuser/svelte-plausible-analytics@"^0.4.0" from the root project
[18:18:54.590] npm ERR! 
[18:18:54.590] npm ERR! Could not resolve dependency:
[18:18:54.590] npm ERR! peer svelte@"^3.59.1" from @accuser/svelte-plausible-analytics@0.4.0
[18:18:54.591] npm ERR! node_modules/@accuser/svelte-plausible-analytics
[18:18:54.591] npm ERR!   @accuser/svelte-plausible-analytics@"^0.4.0" from the root project
[18:18:54.591] npm ERR! 
[18:18:54.591] npm ERR! Fix the upstream dependency conflict, or retry
[18:18:54.591] npm ERR! this command with --force or --legacy-peer-deps
[18:18:54.591] npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
[18:18:54.591] npm ERR! 
[18:18:54.591] npm ERR! 
[18:18:54.591] npm ERR! For a full report see:
[18:18:54.591] npm ERR! /vercel/.npm/_logs/2023-06-22T16_18_52_892Z-eresolve-report.txt
[18:18:54.592] 
[18:18:54.592] npm ERR! A complete log of this run can be found in:
[18:18:54.592] npm ERR!     /vercel/.npm/_logs/2023-06-22T16_18_52_892Z-debug-0.log
[18:18:54.609] Error: Command "npm install" exited with 1
[18:18:55.055] BUILD_UTILS_SPAWN_1: Command "npm install" exited with 1


Best,
Matze

--- a/.eslintignore
+++ b/.eslintignore
@@ -1,0 +1,13 @@
+.DS_Store
+node_modules
+/build
+/.svelte-kit
+/package
+.env
+.env.*
+!.env.example
+
+# Ignore files for PNPM, NPM and YARN
+pnpm-lock.yaml
+package-lock.json
+yarn.lock

--- a/.eslintrc.cjs
+++ b/.eslintrc.cjs
@@ -1,20 +1,30 @@
 module.exports = {
 	root: true,
+	extends: [
+		'eslint:recommended',
+		'plugin:@typescript-eslint/recommended',
+		'plugin:svelte/recommended',
+		'prettier'
+	],
 	parser: '@typescript-eslint/parser',
-	extends: ['eslint:recommended', 'plugin:@typescript-eslint/recommended', 'prettier'],
-	plugins: ['svelte3', '@typescript-eslint'],
-	ignorePatterns: ['*.cjs'],
-	overrides: [{ files: ['*.svelte'], processor: 'svelte3/svelte3' }],
-	settings: {
-		'svelte3/typescript': () => require('typescript')
-	},
+	plugins: ['@typescript-eslint'],
 	parserOptions: {
 		sourceType: 'module',
-		ecmaVersion: 2020
+		ecmaVersion: 2020,
+		extraFileExtensions: ['.svelte']
 	},
 	env: {
 		browser: true,
 		es2017: true,
 		node: true
-	}
+	},
+	overrides: [
+		{
+			files: ['*.svelte'],
+			parser: 'svelte-eslint-parser',
+			parserOptions: {
+				parser: '@typescript-eslint/parser'
+			}
+		}
+	]
 };

--- a/.gitignore
+++ b/.gitignore
@@ -1,9 +1,11 @@
 .DS_Store
 node_modules
-dist
-build
-.svelte-kit
-package
+/build
+/dist
+/.svelte-kit
+/package
 .env
 .env.*
 !.env.example
+vite.config.js.timestamp-*
+vite.config.ts.timestamp-*

--- a/.prettierignore
+++ b/.prettierignore
@@ -1,0 +1,13 @@
+.DS_Store
+node_modules
+/build
+/.svelte-kit
+/package
+.env
+.env.*
+!.env.example
+
+# Ignore files for PNPM, NPM and YARN
+pnpm-lock.yaml
+package-lock.json
+yarn.lock

--- a/.prettierrc
+++ b/.prettierrc
@@ -2,5 +2,8 @@
 	"useTabs": true,
 	"singleQuote": true,
 	"trailingComma": "none",
-	"printWidth": 100
+	"printWidth": 100,
+	"plugins": ["prettier-plugin-svelte"],
+	"pluginSearchDirs": ["."],
+	"overrides": [{ "files": "*.svelte", "options": { "parser": "svelte" } }]
 }

--- a/package.json
+++ b/package.json
@@ -22,40 +22,18 @@
 		"url": "git+https://github.com/accuser/svelte-plausible-analytics.git"
 	},
 	"scripts": {
-		"build": "vite build && pnpm run package",
+		"dev": "vite dev",
+		"build": "vite build && npm run package",
+		"preview": "vite preview",
+		"package": "svelte-kit sync && svelte-package && publint",
+		"prepublishOnly": "npm run package",
+		"test": "playwright test",
 		"check": "svelte-kit sync && svelte-check --tsconfig ./tsconfig.json",
 		"check:watch": "svelte-kit sync && svelte-check --tsconfig ./tsconfig.json --watch",
-		"dev": "vite dev",
-		"format": "prettier --ignore-path .gitignore --write --plugin-search-dir=. .",
-		"lint": "prettier --ignore-path .gitignore --check --plugin-search-dir=. . && eslint --ignore-path .gitignore .",
-		"package": "svelte-kit sync && svelte-package && publint",
-		"prepublishOnly": "pnpm run package",
-		"preview": "vite preview",
-		"test": "playwright test"
+		"test:unit": "vitest",
+		"lint": "prettier --plugin-search-dir . --check . && eslint .",
+		"format": "prettier --plugin-search-dir . --write ."
 	},
-	"devDependencies": {
-		"@playwright/test": "^1.35.1",
-		"@sveltejs/adapter-auto": "2.1.0",
-		"@sveltejs/kit": "1.20.2",
-		"@sveltejs/package": "2.0.2",
-		"@typescript-eslint/eslint-plugin": "^5.60.0",
-		"@typescript-eslint/parser": "^5.60.0",
-		"eslint": "^8.43.0",
-		"eslint-config-prettier": "^8.8.0",
-		"eslint-plugin-svelte3": "^4.0.0",
-		"prettier": "^2.8.8",
-		"prettier-plugin-svelte": "^2.10.1",
-		"svelte": "^3.59.1",
-		"svelte-check": "^3.4.3",
-		"svelte-preprocess": "^5.0.4",
-		"svelte2tsx": "^0.6.15",
-		"tslib": "^2.5.3",
-		"typescript": "~5.1.3",
-		"vite": "4.3.9"
-	},
-	"type": "module",
-	"svelte": "./dist/index.js",
-	"types": "./dist/index.d.ts",
 	"exports": {
 		".": {
 			"types": "./dist/index.d.ts",
@@ -63,10 +41,35 @@
 		}
 	},
 	"files": [
-		"dist"
+		"dist",
+		"!dist/**/*.test.*",
+		"!dist/**/*.spec.*"
 	],
 	"peerDependencies": {
-		"@sveltejs/kit": "^1.20.2",
-		"svelte": "^3.59.1"
-	}
+		"@sveltejs/kit": "^1.20.4",
+		"svelte": "^4.0.0"
+	},
+	"devDependencies": {
+		"@playwright/test": "^1.28.1",
+		"@sveltejs/adapter-auto": "^2.0.0",
+		"@sveltejs/kit": "^1.20.4",
+		"@sveltejs/package": "^2.0.0",
+		"@typescript-eslint/eslint-plugin": "^5.45.0",
+		"@typescript-eslint/parser": "^5.45.0",
+		"eslint": "^8.28.0",
+		"eslint-config-prettier": "^8.5.0",
+		"eslint-plugin-svelte": "^2.30.0",
+		"prettier": "^2.8.0",
+		"prettier-plugin-svelte": "^2.10.1",
+		"publint": "^0.1.9",
+		"svelte": "^4.0.0",
+		"svelte-check": "^3.4.3",
+		"tslib": "^2.4.1",
+		"typescript": "^5.0.0",
+		"vite": "^4.3.6",
+		"vitest": "^0.25.3"
+	},
+	"svelte": "./dist/index.js",
+	"types": "./dist/index.d.ts",
+	"type": "module"
 }

--- a/playwright.config.ts
+++ b/playwright.config.ts
@@ -5,7 +5,8 @@ const config: PlaywrightTestConfig = {
 		command: 'npm run build && npm run preview',
 		port: 4173
 	},
-	testDir: 'tests'
+	testDir: 'tests',
+	testMatch: /(.+\.)?(test|spec)\.[jt]s/
 };
 
 export default config;

--- a/svelte.config.js
+++ b/svelte.config.js
@@ -1,11 +1,9 @@
 import adapter from '@sveltejs/adapter-auto';
-import preprocess from 'svelte-preprocess';
+import { vitePreprocess } from '@sveltejs/kit/vite';
 
 /** @type {import('@sveltejs/kit').Config} */
 const config = {
-	// Consult https://github.com/sveltejs/svelte-preprocess
-	// for more information about preprocessors
-	preprocess: preprocess(),
+	preprocess: vitePreprocess(),
 
 	kit: {
 		adapter: adapter()

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,3 +1,14 @@
 {
-	"extends": "./.svelte-kit/tsconfig.json"
+	"extends": "./.svelte-kit/tsconfig.json",
+	"compilerOptions": {
+		"allowJs": true,
+		"checkJs": true,
+		"esModuleInterop": true,
+		"forceConsistentCasingInFileNames": true,
+		"resolveJsonModule": true,
+		"skipLibCheck": true,
+		"sourceMap": true,
+		"strict": true,
+		"moduleResolution": "NodeNext"
+	}
 }

--- a/vite.config.js
+++ b/vite.config.js
@@ -1,9 +1,9 @@
-// vite.config.js
 import { sveltekit } from '@sveltejs/kit/vite';
+import { defineConfig } from 'vitest/config';
 
-/** @type {import('vite').UserConfig} */
-const config = {
-	plugins: [sveltekit()]
-};
-
-export default config;
+export default defineConfig({
+	plugins: [sveltekit()],
+	test: {
+		include: ['src/**/*.{test,spec}.{js,ts}']
+	}
+});

